crates/swc_ecma_lints/src/rules/duplicate_bindings.rs

```diff
@@ -12,41 +15,62 @@ pub fn duplicate_bindings() -> Box<dyn Rule> {
     visitor_rule(DuplicateBindings::default())
 }
 
+#[derive(Debug, Default, Clone, Copy)]
+struct BindingInfo {
+    span: Span,
+    unique: bool,
+}
+
 #[derive(Debug, Default)]
 struct DuplicateBindings {
-    bindings: AHashMap<Id, Span>,
+    bindings: AHashMap<Id, BindingInfo>,
     type_bindings: AHashSet<Id>,
 
     var_decl_kind: Option<VarDeclKind>,
     is_pat_decl: bool,
+
+    is_module: bool,
 }
 
 impl DuplicateBindings {
     /// Add a binding.
-    fn add(&mut self, id: &Ident, check_for_var_kind: bool) {
-        if check_for_var_kind {
-            if let Some(VarDeclKind::Var) = self.var_decl_kind {
-                return;
-            }
-        }
-
+    fn add(&mut self, id: &Ident, unique: bool) {
         match self.bindings.entry(id.to_id()) {
             Entry::Occupied(mut prev) => {
-                HANDLER.with(|handler| {
-                    handler
-                        .struct_span_err(id.span, "Duplicate binding")
-                        .span_note(*prev.get(), &format!("{} was declared at here", id.sym))
-                        .emit();
-                });
+                if unique || prev.get().unique {
```

Comment:
This is not correct, because `var` can override anything.

Reply:
It's not true in module.
I have tested this locally with the following code.

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
  </head>
  <body>
    <script>
      var x = 1;
      var x = 2;
      function x() {}
      function x() {} // ok
    </script>

    <script type="module">
      var foo = 1;
      var foo = 2; // ok

      function foo1() {}
      function foo1() {} // error
    </script>

    <script type="module">
      var bar = 1;
      function bar() {} // error
    </script>
  </body>
</html>
```

Reply:
Oh... So `three.js` should be treated as a script?

https://github.com/swc-project/swc/runs/5442747097?check_suite_focus=true

Reply:
I agree. It should be treated as script.